### ISSUE TITLE
Set up system for sessions/users

### ISSUE THREAD

**Issue by wickyvinn** [wickyvinn/climbr#5]
| users |
| --- |
| id |
| username |

| sessions |
| --- |
| session_id |
| user_id |
| expired |

a user signs up
they are given a session id with a userid
they set their permanent info - updates the user table
a user logs out, their sessionid is expired.

a user logs in
they are given a session id with a userid
they set their session info, or update their permanent info
every action that they post, the session id gets checked and their user identity is verified.

**Comment by wickyvinn** [comment]
heroku's sample application for node/sessions: https://devcenter.heroku.com/articles/node-sessions

**Comment by wickyvinn** [comment]
this is pretty helpful:
https://stormpath.com/blog/everything-you-ever-wanted-to-know-about-node-dot-js-sessions/

homie seems to wants to do a couple checks:

``` js
app.get('/dashboard', function(req, res) {
  if (req.session && req.session.user) { // Check if session exists
```

not sure if both checks are necessary. ask somebady.

**Comment by wickyvinn** [comment]
Tried:
`connect-mongo`: https://github.com/kcbanner/connect-mongo
`express-mongodb`: https://github.com/biggora/express-mongodb

Comparisons of mongodb session libraries:
http://journal.michaelahlers.org/2013/01/building-with-nodejs-sessions.html

There has been errors about middleware and stuff. Not sure what to do. Gonna try Redis since node seems to handle stuff and it's heroku friendly.

**Comment by wickyvinn** [comment]
http://respectthecode.tumblr.com/post/16167537655/using-mongoose-and-connect-mongodb-with-node-js

**Comment by wickyvinn** [comment]
Fixed by #16
